Plutarch/Deriver/PNewtype.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+
+module Plutarch.Deriver.PNewtype (DerivePNewType (DerivePNewType)) where
+
+import Data.Coerce (Coercible, coerce)
+import Plutarch (PInner, PlutusType, pcon', pmatch', punsafeCoerce)
+import Plutarch.Builtin (PAsData, PIsData, pdata, pfromData)
+import Plutarch.Prelude
+
+{- | Facilitates deriving 'PlutusType' and 'PIsData' for newtypes.
+
+For any newtype represented as-
+> newtype PFoo (s :: S) = PFoo (Term s PBar)
+
+where 'PBar' has a 'PIsData' instance, you can derive 'PlutusType' and 'PIsData' using-
+> deriving (PlutusType, PIsData) via (DerivePNewType PFoo PBar)
+
+This will make 'PFoo' simply be represnted as 'PBar' under the hood.
+-}
+newtype DerivePNewType a (b :: PType) s = DerivePNewType (a s)
```

Comment:
Wrt. naming: `DerivePConstantViaNewtype` is `Newtype` and not `NewType`. We probably want to use the same convention for both. Feel free to choose either.